src/projectM-sdl/pmSDL.cpp

```diff
@@ -426,13 +431,30 @@ void projectMSDL::pollEvent() {
             case SDL_KEYDOWN:
                 keyHandler(&evt);
                 break;
+            case SDL_MOUSEBUTTONDOWN:
+                // Get mouse coorindates when you click.
+                SDL_GetMouseState(&mousex, &mousey);
+                // Scale those coordinates. libProjectM supports a scale of 0.1 instead of absolute pixel coordinates.
+                mousexscale = (mousex/(float)width);
+                mouseyscale = ((height-mousey)/(float)height );
+                // Touch. By not supplying a touch type, we will default to random.
+                touch(mousexscale,mouseyscale,mousepressure);
+                break;
             case SDL_QUIT:
                 done = true;
                 break;
         }
     }
 }
 
+void projectMSDL::touch(float x, float y, int pressure, int touchtype) {
+    if (y > 1)
+        y = 1.0;
```

Comment:
what's this check for? is it possible?
comment this method

Reply:
It's not possible anymore. I wrote that when I made a mistake and was going math off the wrong axis and didn't understand what I was getting a positive value. I'll remove it now.